pkg/vmimage/data/script.sh

```diff
@@ -11,8 +11,15 @@ xfs_growfs /dev/sda2
 yum -y update -x WALinuxAgent # updating WALinuxAgent kills this script
 yum -y install git libguestfs-tools-c libvirt-daemon-config-network virt-install
 
+CHECKSUM=14ec881815eb9e6618f95df5eb385d961283efc196d97912595ba6484a56180d
 # install golang-1.11 from source because rhel7 ships with golang 1.9 by default
-curl -s https://storage.googleapis.com/golang/go1.11.12.linux-amd64.tar.gz | tar -C /usr/local -xz
+curl -s https://storage.googleapis.com/golang/go1.11.12.linux-amd64.tar.gz --output /tmp/go.1.11.12.linux.amd64.tar.gz
+if [[ $(sha256sum /tmp/go.1.11.12.linux-amd64.tar.gz | awk '{print $1}') == $CHECKSUM ]]; then
```

Comment:
Might be simpler to use `sha256sum`'s `-c` flag here:

```suggestion
echo "14ec881815eb9e6618f95df5eb385d961283efc196d97912595ba6484a56180d  /tmp/go.1.11.12.linux-amd64.tar.gz" | sha256sum -c
```

Note that since we've `set -x`, if this errors out, then we will automatically exit (and don't need to include the if-else).

Reply:
NICE!! I like the -c option, so much easier.

Reply:
This makes sense, the `-x` will automatically exit.